polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -591,6 +592,8 @@ impl<T: Config> Pallet<T> {
 
 		METRICS.on_candidates_processed_total(backed_candidates.len() as u64);
 
+		filter_elastic_scaling_candidates::<T>(&mut backed_candidates);
```

Comment:
Ok, we need to clean up this function more. We should ensure at the top-level at if [on enter](https://github.com/paritytech/polkadot-sdk/blob/a5ba15722208e0aeeeff3ca0dd161bedc31a8b1d/polkadot/runtime/parachains/src/paras_inherent/mod.rs#L111) anything is filtered, then we should err out like [here](https://github.com/paritytech/polkadot-sdk/blob/a5ba15722208e0aeeeff3ca0dd161bedc31a8b1d/polkadot/runtime/parachains/src/paras_inherent/mod.rs#L487), as the filtering should have happened on create_inherent already. Otherwise it is possible to produce blocks with lots of garbage (occupying storage) not having any effect.

Reply:
Correct, I added this